hail/python/hail/methods/qc.py

```diff
@@ -491,13 +504,128 @@ def n_discordant(counter):
+            local_env['VEP_OUTPUT_FILE'] = local_output_file
+
+            bb.create_job(image,
+                          csq_command,
+                          attributes={'name': 'csq-header'},
+                          resources={'cpu': '1', 'memory': 'standard'},
+                          cloudfuse=[(data_bucket, data_mount, True)],
+                          output_files=[(local_output_file, f'{vep_output_path}/csq-header')],
+                          regions=[region],
+                          requester_pays_project=requester_pays_project,
+                          env=local_config['env'],
+                          )
+
+        for f in hl.hadoop_ls(vep_input_path):
+            path = f['path']
+            part_name = os.path.basename(path)
+            if not part_name.startswith('part-'):
+                continue
+            part_id = int(part_name.split('-')[1])
+
+            run_vep_command = local_config['command']
+
+            local_input_file = '/io/input'
+            local_output_file = '/io/output.gz'
+
+            local_env = copy.deepcopy(env)
+            local_env['VEP_BLOCK_SIZE'] = str(block_size)
+            local_env['VEP_PART_ID'] = str(part_id)
+            local_env['VEP_DATA_MOUNT'] = shq(data_mount)
+            local_env['VEP_CONSEQUENCE'] = str(int(csq))
+            local_env['VEP_TOLERATE_PARSE_ERROR'] = str(int(tolerate_parse_error))
+            local_env['VEP_INPUT_FILE'] = local_input_file
+            local_env['VEP_OUTPUT_FILE'] = local_output_file
+
+            bb.create_job(image,
+                          run_vep_command,
+                          attributes={'name': f'vep-{part_id}'},
+                          resources={'cpu': '1', 'memory': 'standard'},
+                          input_files=[(path, local_input_file)],
+                          output_files=[(local_output_file, f'{vep_output_path}/annotations/{part_name}.tsv.gz')],
+                          cloudfuse=[(data_bucket, data_mount, True)],
+                          regions=[region],
+                          requester_pays_project=requester_pays_project,
+                          env=local_env,
+                          )
+
+    async_to_blocking(backend._submit_batch(
+        build_vep_batch, 'vep(...)', attributes={'vep': '1', 'token': token}, cancel_after_n_failures=1
+    ))
+
+    annotations = hl.import_table(f'{vep_output_path}/annotations/*',
+                                  key='variant',
+                                  types={'variant': hl.tstr,
+                                         'vep': vep_typ,
+                                         'vep_proc_id': hl.tstruct(part_id=hl.tint,
+                                                                   block_id=hl.tint)},
+                                  force=True)
+
+    reference_genome = ht.locus.dtype.reference_genome.name
+    annotations = annotations.key_by(**hl.parse_variant(annotations.variant, reference_genome=reference_genome))
```

Comment:
@tpoterba is it possible to avoid this key_by by specifying the right kind of type in `types`?

Reply:
not possible. However, it's incorrect to use `key='variant'` in import_table -- that should get optimized away, but as written this keys the table by the variant string, which is definitely not sorted.